### ISSUE TITLE
SAM Deploy fails when using InlineCode

### ISSUE THREAD

**Issue by keetonian** [awslabs/aws-sam-cli#1644]
### Description

`sam deploy` doesn't work when using InlineCode. I use `sam deploy` with InlineCode to quickly test items and avoid any extra time spent on package/deploy steps.

### Steps to reproduce

Template:

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Resources:
  MyFunction2:
    Type: AWS::Serverless::Function
    Properties:
      InlineCode: |
        def handler(event, context):
            print(event)
            return {'body': 'Hello World 2!', 'statusCode': 200}
      Handler: index.handler
      Runtime: python3.7
      Events:
        Api:
          Type: Api
          Properties:
            Path: /
            Method: get
```

Command: 

```
sam deploy --template-file template.yaml --stack-name testing-inline-api --capabilities CAPABILITY_IAM
```
### Observed result

```
Unable to export
Traceback (most recent call last):
  File "/usr/local/Cellar/aws-sam-cli/0.35.0/libexec/lib/python3.7/site-packages/samcli/lib/package/artifact_exporter.py", line 255, in export
    self.do_export(resource_id, resource_dict, parent_dir)
  File "/usr/local/Cellar/aws-sam-cli/0.35.0/libexec/lib/python3.7/site-packages/samcli/lib/package/artifact_exporter.py", line 271, in do_export
    uploaded_url = upload_local_artifacts(resource_id, resource_dict, self.PROPERTY_NAME, parent_dir, self.uploader)
  File "/usr/local/Cellar/aws-sam-cli/0.35.0/libexec/lib/python3.7/site-packages/samcli/lib/package/artifact_exporter.py", line 151, in upload_local_artifacts
    return zip_and_upload(local_path, uploader)
  File "/usr/local/Cellar/aws-sam-cli/0.35.0/libexec/lib/python3.7/site-packages/samcli/lib/package/artifact_exporter.py", line 162, in zip_and_upload
    return uploader.upload_with_dedup(zip_file)
  File "/usr/local/Cellar/aws-sam-cli/0.35.0/libexec/lib/python3.7/site-packages/samcli/lib/package/s3_uploader.py", line 127, in upload_with_dedup
    return self.upload(file_name, remote_path)
  File "/usr/local/Cellar/aws-sam-cli/0.35.0/libexec/lib/python3.7/site-packages/samcli/lib/package/s3_uploader.py", line 76, in upload
    if not self.force_upload and self.file_exists(remote_path):
  File "/usr/local/Cellar/aws-sam-cli/0.35.0/libexec/lib/python3.7/site-packages/samcli/lib/package/s3_uploader.py", line 140, in file_exists
    raise BucketNotSpecifiedError()
samcli.commands.package.exceptions.BucketNotSpecifiedError: 
S3 Bucket not specified, use --s3-bucket to specify a bucket name or run sam deploy --guided
Sending Telemetry: {'metrics': [{'commandRun': {'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam deploy', 'duration': 452, 'exitReason': 'ExportFailedError', 'exitCode': 1, 'requestId': 'e1d1b60a-60e8-4e58-808c-c25cc33d35e3', 'installationId': 'bf6f64ec-101c-44d8-b441-aba84b49340b', 'sessionId': 'a3844b89-d64a-4d17-a228-e13991378cf1', 'executionEnvironment': 'CLI', 'pyversion': '3.7.5', 'samcliVersion': '0.35.0'}}]}
Telemetry response: 200
Error: Unable to upload artifact None referenced by CodeUri parameter of MyFunction2 resource.

S3 Bucket not specified, use --s3-bucket to specify a bucket name or run sam deploy --guided
```

### Expected result

I should be able to deploy without referencing a S3 bucket. I don't use one in my deployment (`InlineCode`), so why do I need to reference one? The error readout even says "`Unable to upload artifact *None*`"

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: OSX
2. `sam --version`: 0.35.0

`Add --debug flag to command you are running`

**Comment by keetonian** [comment]
It works if I use `--s3-bucket` in the command, but that requires me to set up a bucket in any region I want to deploy to.

**Comment by sanathkr** [comment]
oh this is a bug. @TheSriram we need to fix this

**Comment by gdw2** [comment]
Looks like I duped it: 
https://github.com/awslabs/serverless-application-model/issues/1309

**Comment by jfuss** [comment]
Patched was released in [0.38.0](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.38.0).

Closing
